src/view/treeNodes/pullRequestNode.ts

```diff
@@ -346,96 +284,175 @@ export class PRNode extends TreeNode implements CommentHandler, vscode.Commentin
+			return;
+		}
+
+		const reviewThreads = this.pullRequestModel.reviewThreadsCache;
+		const threadsByPath = groupBy(reviewThreads, thread => thread.path);
 
-		for (const fileName in commentThreadCache) {
-			const commentThreads = commentThreadCache[fileName];
+		const commentThreadCache = (await this.resolvePRCommentController()).commentThreadCache;
 
-			const matchedEditor = currentPRDocuments.find(editor => editor.fileName === fileName);
+		const prEditors = this.getPREditors(editors);
+		this._openPREditors = prEditors;
+		prEditors.forEach(editor => {
+			const { fileName, isBase } = fromPRUri(editor.document.uri);
+			if (threadsByPath[fileName]) {
+				commentThreadCache[fileName] = threadsByPath[fileName]
+					.filter(
+						thread =>
+							(thread.diffSide === DiffSide.LEFT && isBase) ||
+							(thread.diffSide === DiffSide.RIGHT && !isBase),
+					)
+					.map(thread => {
+						const range = new vscode.Range(
+							new vscode.Position(thread.line - 1, 0),
+							new vscode.Position(thread.line - 1, 0),
+						);
 
-			if (!matchedEditor) {
-				commentThreads.forEach(thread => thread.dispose!());
-				delete commentThreadCache[fileName];
+						return createVSCodeCommentThreadForReviewThread(
+							editor.document.uri,
+							range,
+							thread,
+							this._commentController,
+						);
+					});
 			}
-		}
+		});
 
-		if (!incremental) {
-			// it's triggered by file opening, so we only take care newly opened documents.
-			currentPRDocuments = currentPRDocuments.filter(editor => commentThreadCache[editor.fileName] === undefined);
-		}
+		this._hasInitializedThreads = true;
+	}
 
-		currentPRDocuments = uniqBy(currentPRDocuments, editor => editor.fileName);
+	private onDidChangeOpenEditors(editors: vscode.TextEditor[]): void {
+		const commentThreadCache = this._prCommentController!.commentThreadCache;
+		const prEditors = this.getPREditors(editors);
+		const removed = this._openPREditors.filter(x => !prEditors.includes(x));
+		const added = prEditors.filter(x => !this._openPREditors.includes(x));
+		this._openPREditors = prEditors;
 
-		if (currentPRDocuments.length) {
-			const fileChanges = await this.getFileChanges();
-			await this.pullRequestModel.validateDraftMode();
-			currentPRDocuments.forEach(editor => {
-				const fileChange = fileChanges.find(fc => fc.fileName === editor.fileName);
+		removed.forEach(editor => {
+			const fileName = fromPRUri(editor.document.uri)!.fileName;
+			const threads = commentThreadCache[fileName] || [];
+			threads.forEach(t => t.dispose());
+			delete commentThreadCache[fileName];
+		});
 
-				if (!fileChange || fileChange instanceof RemoteFileChangeNode) {
-					return;
+		if (added.length) {
+			const reviewThreads = this.pullRequestModel.reviewThreadsCache;
+			const threadsByPath = groupBy(reviewThreads, thread => thread.path);
+			added.forEach(editor => {
+				const { isBase, fileName } = fromPRUri(editor.document.uri);
+				if (threadsByPath[fileName]) {
```

Comment:
Is this the same as line 317 above? Might be nice to factor it into a sepearte function.

Reply:
Thanks, moved!